vmdb/db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Description: This method checks to see if the stack has been provisioned
+#   and whether the refresh has completed
+#
+
+def refresh_provider(service)
+  provider = service.orchestration_manager
+
+  $evm.log("info", "Refreshing provider #{provider.name}")
+  provider.refresh
+end
+
+$evm.log("info", "Check orchestration provisioned")
+
+service = $evm.root["service_template_provision_task"].destination
+
+if $evm.state_var_exist? 'stack_deployed'
+  # check whether refresh completed
+  stack_name = service.stack_name
+  provider = service.orchestration_manager
+  stack = $evm.vmdb('orchestration_stack').find_by_name_and_ems_id(stack_name, provider.id)
```

Comment:
Can we store the stack_id from deploy_orchestration_stack and use it for the lookup here instead of relying on the name?